Source/DafnyCore/Compilers/Compiler-Csharp.cs

```diff
@@ -3390,7 +3390,7 @@ private class CSharpCompilationResult {
     }
 
     private void AddTestCheckerIfNeeded(string name, Declaration decl, ConcreteSyntaxTree wr) {
-      if (DafnyOptions.O.RunAllTests || !Attributes.Contains(decl.Attributes, "test")) {
+      if (DafnyOptions.O.Compile || DafnyOptions.O.RunAllTests || !Attributes.Contains(decl.Attributes, "test")) {
```

Comment:
Just checking that the intended behavior is that methods marked {:test} are only compiled when testing

Reply:
They're always compiled when compiling, but the XUnit attribute is only added when using `dafny translate`.